src/arb/ComplexPoly.jl

```diff
@@ -731,20 +722,65 @@ function setcoeff!(z::ComplexPolyRingElem, n::Int, x::ComplexFieldElem)
   return z
 end
 
+#
+
+function add!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::ComplexPolyRingElem)
+  ccall((:acb_poly_add, libflint), Nothing,
+        (Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Int),
+        z, x, y, precision(Balls))
+  return z
+end
+
+function add!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::Int)
+  ccall((:acb_poly_add_si, libflint), Nothing,
+        (Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Int, Int),
+        z, x, y, precision(Balls))
+  return z
+end
+
+add!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::ComplexFieldElem) = add!(z, x, parent(z)(y))
+
+add!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::ZZRingElem) = add!(z, x, parent(z)(y))
+
+add!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::Integer) = add!(z, x, flintify(y))
+
+add!(z::ComplexPolyRingElem, x::Union{ComplexFieldElem,IntegerUnion}, y::ComplexPolyRingElem) = add!(z, y, x)
+
+#
+
+function sub!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::ComplexPolyRingElem)
+  ccall((:acb_poly_sub, libflint), Nothing,
+        (Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Int),
+        z, x, y, precision(Balls))
+  return z
+end
+
+sub!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::Union{ComplexFieldElem,IntegerUnion}) = sub!(z, x, parent(z)(y))
+
+sub!(z::ComplexPolyRingElem, x::Union{ComplexFieldElem,IntegerUnion}, y::ComplexPolyRingElem) = sub!(z, parent(z)(x), y)
+
+#
+
 function mul!(z::ComplexPolyRingElem, x::ComplexPolyRingElem, y::ComplexPolyRingElem)
   ccall((:acb_poly_mul, libflint), Nothing,
         (Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Ref{ComplexPolyRingElem}, Int),
-        z, x, y, precision(parent(z)))
+        z, x, y, precision(Balls))
```

Comment:
Why this change?

Reply:
All existing arithmetic used the global precision, including `*`. So when unifying `*` and `mul!` I had to decide on one and went with the dozens of times used one instead of `precision(parent(z))` which was only used in these three methods (and with this PR nowhere)

Reply:
OK. It most likely is a copy&paste artifact anyway.